sunxspex/xspec_models.py

```diff
@@ -0,0 +1,367 @@
+        df=pd.DataFrame({h:r for h,r in zip(headers,partable)}, index=range(1,len(self.ParInfo)+1))
+        return df #table should be small enough to print well in both terminal and Jupyter
+
+    def other_method(self):
+        '''such as: easily set parameter defaults from a tuple or dictionary, descriptions of parameters, etc'''
+        raise NotImplementedError
+
+class ThickTargetModel(XspecModel):
+    '''Thick-target bremsstrahlung model for use in Xspec. Default parameters are taken from OSPEX [link]()'''
+    def __init__(self):
+        self.ParInfo=(
+        "a0  1e-35  100.0  1.0  10.0  1e6  1e7  1.0",
+        "p  \"\"  4.0  1.1  1.5  15.0  20.0  0.1",
+           "eebrk  keV  150.0  1.0  5.0  100.  1e5  0.5" ,
+         "q  \"\"  6.0  0.0  1.5  15.0  20.0  0.1",
+         "eelow  keV  20.0  0.0  1.0  100.  1e3  1.0" ,
+         "eehigh  keV  3200.0  1.0  10.0  1e6  1e7  1.0"
+        )
+        self.model=self.thick2
+        self.description=f"Thick-target bremsstrahlung '{self.model.__name__}'"
+
+    @staticmethod #XSPEC won't know what to do with self as first arg
+    def thick2(photon_energies, params, flux):
+        """
+        tuples or lists containing photon energies, params
+              flux is empty list of length nE-1
+
+              The input array of energy bins gives the boundaries of the energy bins
+              and hence has one more entry than the output flux arrays.
+
+              The output flux array for an additive model should be in terms of photons/cm$^2$/s
+              (not photons/cm$^2$/s/keV) i.e. it is the model spectrum integrated over the energy bin.
+        """
+        photon_energies=np.array(photon_energies)
+        photon_energy_bins=photon_energies[1:]-photon_energies[:-1]
+        photon_energy_bin_centers=(photon_energies[1:]-photon_energies[:-1])/2+photon_energies[:-1] #assumes linear spacing...
+        internal_flux=np.zeros(photon_energies.shape)
+
+        # Constants
+        #const=constants.Constants() #assume this has already happened
+        mc2 = const.get_constant('mc2')
+        clight = const.get_constant('clight')
+        au = const.get_constant('au')
+        r0 = const.get_constant('r0')
+
+        # Max number of points
+        maxfcn = 2048
+
+        # Average atomic number
+        z = 1.2
+
+        # Relative error
+        rerr = 1e-4
+
+        # Numerical coefficient for photo flux
+        fcoeff = ((clight ** 2 / mc2 ** 4) / (4 * np.pi * au ** 2))
+        decoeff = 4.0 * np.pi * (r0 ** 2) * clight
+
+        # Create arrays for the photon flux and error flags.
+        iergq = np.zeros_like(photon_energies, dtype=np.float64)
+        try:
+            a0,p, eebrk, q, eelow, eehigh,_=params #why is this not consistent within xspec?
+        except ValueError:
+            a0,p, eebrk, q, eelow, eehigh=params
+
+        #if eelow >= eehigh:
+        #    return list(photon_energies)[1:]
```

Comment:
Is this still needed?  If not, we should just delete it.

Reply:
have to check, it might speed things up in that extreme case